tests/resources/serializers/test_marcxml_serializer.py

```diff
@@ -136,115 +146,38 @@ def test_marcxml_serializer_minimal_record(running_app, minimal_record, parent):
     assert serialized_record == expected_data
 
 
-def _add_file_to_record(recid, client, headers):
-    """Adds a file to the record."""
-    # Attach a file to it
-    response = client.post(
-        f"/records/{recid}/draft/files", headers=headers, json=[{"key": "test.pdf"}]
-    )
-    assert response.status_code == 201
-    response = client.put(
-        f"/records/{recid}/draft/files/test.pdf/content",
-        headers={
-            "content-type": "application/octet-stream",
-            "accept": "application/json",
-        },
-        data=BytesIO(b"testfile"),
-    )
-    assert response.status_code == 200
-    response = client.post(
-        f"/records/{recid}/draft/files/test.pdf/commit", headers=headers
-    )
-    assert response.status_code == 200
-
-
-def _add_record_to_communities(db, recid, community, community2):
-    """Add record to communities."""
-    record = current_rdm_records_service.read(
-        id_=recid, identity=system_identity
-    )._record
-
-    record.parent.communities.add(community._record, default=True)
-    record.parent.communities.add(community2._record, default=False)
-    record.parent.commit()
-    db.session.commit()
-    current_rdm_records_service.indexer.index(record, arguments={"refresh": True})
-
-
-def test_marcxml_serializer_full_record(
-    db,
-    running_app,
-    updated_full_record,
-    client_with_login,
-    headers,
-    community,
-    community2,
-):
-    """Test serializer for MARCXML with a"""
+def test_marcxml_serializer_full_record(db, running_app, updated_full_record):
```

Comment:
I refactored the test to follow other serializer test patterns: input a static dictionary, serialize it and compare the result.

The previous way of testing this serializer made tests fail sometimes because of the order of the record's communities which was not deterministic.